esmvalcore/preprocessor/_volume.py

```diff
@@ -125,33 +126,52 @@ def calculate_volume(cube: Cube) -> da.core.Array:
     """
     # Load depth field and figure out which dim is which
     depth = cube.coord(axis='z')
-    z_dim = cube.coord_dims(depth)[0]
+    z_dim = cube.coord_dims(depth)
```

Comment:
```suggestion
    z_dim = cube.coord_dims(depth)
    depth = depth.copy()
```

To avoid overwriting the input cube